plugins/catalog-backend/src/ingestion/processors/GithubDiscoveryProcessor.ts

```diff
@@ -78,16 +78,14 @@ export class GithubDiscoveryProcessor implements CatalogProcessor {
     this.logger.info(`Reading GitHub repositories from ${location.target}`);
 
     const { repositories } = await getOrganizationRepositories(client, org);
+    const matching = repositories.filter(r => repoSearchPath.test(r.name));
 
     const duration = ((Date.now() - startTimestamp) / 1000).toFixed(1);
     this.logger.debug(
-      `Read ${repositories.length} GitHub repositories in ${duration} seconds`,
+      `Read ${repositories.length} GitHub repositories (${matching.length} matching the pattern) in ${duration} seconds`,
     );
 
-    for (const repository of repositories) {
-      if (!repoSearchPath.test(repository.name)) {
-        continue;
-      }
+    for (const repository of matching) {
```

Comment:
Looks like not just a log improvement, but a performance bug fix too yes? (no point reviewing the 47 non matched repos for the repo name?) 🎉

Reply:
Nah it's functionally the same - the `continue` skipped over all non-matching things, I just do it outside the loop instead (so I get the count sooner)